cmd/util/flags/cliflags/job.go

```diff
@@ -0,0 +1,232 @@
+			// We set only the key here.
+			rawLabels[l] = ""
+		}
+	}
+	for k, v := range rawLabels {
+		if !IsSafeAnnotation(k) || k == "" {
+			return nil, fmt.Errorf("label with key %q is not allowed", k)
+		}
+		if !IsSafeAnnotation(v) || v == "" {
+			return nil, fmt.Errorf("label with value %q is not allowed", k)
+		}
+	}
+
+	return rawLabels, nil
+}
+
+type LegacyJobFlags struct {
+	// Deprecated: use `JobSettings.jobType`
+	targetingMode model.TargetingMode
+	// Deprecated: use 'JobSettings.labels'
+	labels []string
+	// Deprecated: use 'JobSettings.constraints'
+	selectors string
+	//Deprecated: use `JobSettings.count`
+	concurrency int
+}
+
+func DefaultJobSettings() *JobSettings {
+	return &JobSettings{
+		name:        "main",
+		namespace:   "default",
+		jobType:     models.JobTypeBatch,
+		priority:    0,
+		count:       1,
+		constraints: "",
+		labels:      make(map[string]string),
+
+		legacy: &LegacyJobFlags{
+			targetingMode: model.TargetAny,
+			labels:        []string{},
+			selectors:     "",
+			concurrency:   1,
+		},
+	}
+}
+
+func RegisterJobFlags(cmd *cobra.Command, s *JobSettings) {
+	s.cmd = cmd
+	fs := pflag.NewFlagSet("job", pflag.ContinueOnError)
+	fs.StringVar(&s.name, "name", s.name,
+		`The name to refer to this job by.`)
+
+	fs.StringVar(&s.namespace, "namespace", s.namespace, `The namespace to associate with this job.`)
+
+	fs.IntVar(&s.priority, "priority", s.priority, `The priority of the job.`)
+
+	//
+	// Deprecation of legacy flags tracked via https://github.com/bacalhau-project/bacalhau/issues/3838
+	//
+
+	// NB(forrest): the `count` flag is replacing `concurrency`. Hide the `concurrency` flag and add deprecation notice.
+	fs.IntVar(&s.count, "count", s.count, `How many nodes should run the job.`)
+
+	fs.IntVar(&s.legacy.concurrency, "concurrency", s.legacy.concurrency,
+		`How many nodes should run the job`)
+
+	if err := fs.MarkHidden("concurrency"); err != nil {
+		panic(err)
+	}
+	if err := fs.MarkDeprecated("concurrency", "use --count"); err != nil {
+		panic(err)
+	}
+
+	// NB(forrest): the `type` flag is replacing `targeting`. Hide the `targeting` flag and add deprecation notice.
+	fs.StringVar(&s.jobType, "type", s.jobType,
+		`The type of the job (batch, ops, system, or daemon).`)
+
+	// deprecated
+	fs.Var(flags.TargetingFlag(&s.legacy.targetingMode), "target",
+		`Whether to target the minimum number of matching nodes ("any") (default) or all matching nodes ("all").`)
+
+	if err := fs.MarkHidden("target"); err != nil {
+		panic(err)
+	}
+	if err := fs.MarkDeprecated("target", "use --type"); err != nil {
+		panic(err)
+	}
+
+	// NB(forrest): the `label` flag is replacing `labels` flag. Hide the `labels` flag and add deprecation notice.
+	fs.StringToStringVar(&s.labels, "label", s.labels,
+		`List of labels for the job. Enter multiple in the format '-label a -label 2'.
+All characters not matching /a-zA-Z0-9_:|-/ and all emojis will be stripped.`)
```

Comment:
we are using plural format for similar flags, including `--labels` with `bacalhau serve`

Reply:
see https://github.com/bacalhau-project/bacalhau/pull/4020#discussion_r1619426801